internal/pkg/policy/parsed_policy.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package policy
+
+import (
+	"encoding/json"
+
+	"github.com/elastic/fleet-server/v7/internal/pkg/model"
+	"github.com/elastic/fleet-server/v7/internal/pkg/smap"
+)
+
+type RoleT struct {
+	Raw  []byte
+	Sha2 string
+}
+
+type RoleMapT map[string]RoleT
+
+type ParsedPolicy struct {
+	Policy model.Policy
+	Fields map[string]json.RawMessage
+	Roles  RoleMapT
+}
+
+func NewParsedPolicy(p model.Policy) (*ParsedPolicy, error) {
```

Comment:
Should add unit tests for this function so we can ensure that a `model.Policy` converts to a `ParsedPolicy` correctly.

Reply:
Will do.